### ISSUE TITLE
unset zle_bracketed_paste when $TERM == "dumb" please

### ISSUE THREAD

**Issue by scinart** [sorin-ionescu/prezto#1552]
### Description
zsh is not working with Emacs tramp

It is because of the bracketed paste thing, see https://emacs.stackexchange.com/questions/19848/weird-characters-in-shell-mode-with-zsh/19853

Would you please consider inserting

    if [[ $TERM = dumb ]]; then
      unset zle_bracketed_paste
    fi

somewhere in the config file?

**Comment by belak** [comment]
Interesting find.

Out of curiosity, do you still see this issue if the `environment` module is disabled? I think a better fix might be to only enable bracketed paste mode if the terminal is not dumb, but I'm not sure that'll fix it.

**Comment by scinart** [comment]
Yes.

You can reproduce this behavior by

    TERM=dumb ssh user@ip -t | xxd
    # enter the password
    # interactive shell but piped to xxd, enter "exit⏎". 

You will see some `[?2004h` after the prompt. It is not visible unless by using `xxd`.

**Comment by belak** [comment]
I've added some code to only enable bracketed paste if you're not in a dumb terminal. Might be worth doing the same thing for non-interactive terminals as well at some point.

Thanks for filing this!

**Comment by sfc-gh-sparkes** [comment]
> I've added some code to only enable bracketed paste if you're not in a dumb terminal. Might be worth doing the same thing for non-interactive terminals as well at some point.

Just an FYI:

I noticed today that `zstyle`ing `color` to `yes` as is done by default in `zpreztorc` causes `zle_bracketed_paste` to get set as a side effect, at least in `5.8`. Setting `color` to `no` later in `zprezto/init.zsh` does _not_ unset `zle_bracketed_paste`.

I tried guarding the `zstyle` at the beginning of `zpreztorc` but other modules set `color` with the result that `zle_bracketed_paste` gets set. I didn't track down which module(s) are doing this.

Instead, I took the approach of unsetting `zle_bracketed_paste` in `zprezto/init.zsh` where `color` is disabled based on TERM type.
